include/gsl/gsl

```diff
@@ -74,69 +74,63 @@ class not_null
     static_assert(std::is_assignable<T&, std::nullptr_t>::value, "T cannot be assigned nullptr.");
 
 public:
-    not_null(T t) : ptr_(t) { ensure_invariant(); }
-    not_null& operator=(const T& t)
+    template <typename U, typename Dummy = std::enable_if_t<std::is_convertible<U, T>::value>>
+    constexpr not_null(U&& u) : ptr_(std::forward<U>(u)) { ensure_invariant(); }
+
+    template <typename U, typename Dummy = std::enable_if_t<std::is_convertible<U, T>::value>>
+    constexpr not_null(const not_null<U>& other) : ptr_(other.get())
     {
-        ptr_ = t;
-        ensure_invariant();
-        return *this;
+        ensure_invariant(); // Just in case a non-null U converts to null T.
     }
 
     not_null(const not_null& other) = default;
     not_null& operator=(const not_null& other) = default;
 
-    template <typename U, typename Dummy = std::enable_if_t<std::is_convertible<U, T>::value>>
-    not_null(const not_null<U>& other)
+    constexpr T get() const
     {
-        *this = other;
+        // help the optimizer
+        GSL_ASSUME(ptr_ != nullptr);
+        return ptr_;
     }
 
-    template <typename U, typename Dummy = std::enable_if_t<std::is_convertible<U, T>::value>>
-    not_null& operator=(const not_null<U>& other)
-    {
-        ptr_ = other.get();
-        return *this;
-    }
+    constexpr operator T() const { return get(); }
+    constexpr T operator->() const { return get(); }
 
-    // prevents compilation when someone attempts to assign a nullptr
-    not_null(std::nullptr_t) = delete;
-    not_null(int) = delete;
-    not_null<T>& operator=(std::nullptr_t) = delete;
-    not_null<T>& operator=(int) = delete;
+    // unary +: because there's no particular reason to forbid it,
+    // and it's sometimes useful to convert an lvalue to a prvalue.
```

Comment:
No, the only "weirdness" here was that my pass looking for unintended arithmetic operators happened long enough after my pass to remove unneeded operator overloads that I didn't realize that unary `+` didn't need an overload.

I'm interpreting the consensus here as "remove this useless overload but don't `delete` unary `+`."